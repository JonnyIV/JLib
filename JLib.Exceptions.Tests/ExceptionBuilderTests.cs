using FluentAssertions;
using Xunit;

namespace JLib.Exceptions.Tests;

public class ExceptionBuilderTests
{
    private readonly ExceptionBuilder _builder = new("test");
    [Fact]
    public void Test1()
        => _builder.HasErrors().Should().BeFalse();

    [Fact]
    public void Test2()
    {
        _builder.CreateChild("child");
        _builder.HasErrors().Should().BeFalse();
    }

    [Fact]
    public void Test3()
    {
        var child = _builder.CreateChild("child");
        child.Add(new Exception("error"));
        _builder.HasErrors().Should().BeTrue();
    }
    [Fact]
    public void Test4()
    {
        var child = _builder.CreateChild("child");
        _builder.Add(new Exception("error"));
        _builder.HasErrors().Should().BeTrue();
    }
}