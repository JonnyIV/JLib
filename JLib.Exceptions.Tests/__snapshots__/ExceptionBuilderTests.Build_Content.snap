{
  "TargetSite": null,
  "Message": "{\r\n  \u0022Message\u0022: \u0022test\u0022,\r\n  \u0022Exception\u0022: \u0022content\u0022\r\n}",
  "Data": {},
  "InnerException": {
    "TargetSite": null,
    "Message": "content",
    "Data": {},
    "InnerException": null,
    "HelpLink": null,
    "Source": null,
    "HResult": -2146233088,
    "StackTrace": null
  },
  "HelpLink": null,
  "Source": null,
  "HResult": -2146233088,
  "StackTrace": null
}
